distributed/utils.py

```diff
@@ -1994,3 +1995,11 @@ def __eq__(self, other):
 
     def __lt__(self, other):
         return self.obj < other.obj
+
+
+@functools.lru_cache
+def url_escape(url, *args, **kwargs):
+    """
+    Escape a URL path segment.
```

Comment:
Non-blocking suggestion for future codebase readers 

```suggestion
    Escape a URL path segment. Cache results for better performance. 
```